pkg/migration-tool/writer/writer.go

```diff
@@ -21,44 +22,51 @@ const (
 	backOffRetryDuration = time.Millisecond * 100
 )
 
-type RemoteWrite struct {
-	c                  context.Context
-	sigSlabRead        chan *planner.Slab
-	url                string
+// Config is config for writer.
+type Config struct {
+	Context          context.Context
+	Url              string
+	MigrationJobName string // Label value to the progress metric.
+	ConcurrentPush   int
+	HTTPConfig       config.HTTPClientConfig
+
+	ProgressEnabled    bool
+	ProgressMetricName string // Metric name to main the last pushed maxt to remote write storage.
+
+	SigSlabRead chan *planner.Slab
+	SigSlabStop chan struct{}
+}
+
+type Write struct {
+	Config
 	shardsSet          *shardsSet
-	slabsPushed        atomic.Int64
-	progressMetricName string // Metric name to main the last pushed maxt to remote write storage.
-	migrationJobName   string // Label value to the progress metric.
+	slabsPushed        int64
 	progressTimeSeries *prompb.TimeSeries
 }
 
 // New returns a new remote write. It is responsible for writing to the remote write storage.
-func New(c context.Context, remoteWriteUrl, progressMetricName, migrationJobName string, numShards int, progressEnabled bool, sigRead chan *planner.Slab) (*RemoteWrite, error) {
-	ss, err := newShardsSet(c, remoteWriteUrl, numShards)
+func New(config Config) (*Write, error) {
+	ss, err := newShardsSet(config.Context, config.HTTPConfig, config.Url, config.ConcurrentPush)
 	if err != nil {
 		return nil, fmt.Errorf("creating shards: %w", err)
 	}
-	write := &RemoteWrite{
-		c:                  c,
-		shardsSet:          ss,
-		url:                remoteWriteUrl,
-		sigSlabRead:        sigRead,
-		migrationJobName:   migrationJobName,
-		progressMetricName: progressMetricName,
+	write := &Write{
+		Config:    config,
+		shardsSet: ss,
 	}
-	if progressEnabled {
+	if config.ProgressEnabled {
 		write.progressTimeSeries = &prompb.TimeSeries{
-			Labels: utils.LabelSet(progressMetricName, migrationJobName),
+			Labels: utils.LabelSet(config.ProgressMetricName, config.MigrationJobName),
 		}
 	}
-	write.slabsPushed.Store(0)
+	atomic.StoreInt64(&write.slabsPushed, 0)
 	return write, nil
 }
 
 // Run runs the remote-writer. It waits for the remote-reader to give access to the in-memory
 // data-block that is written after the most recent fetch. After reading the block and storing
 // the data locally, it gives back the writing access to the remote-reader for further fetches.
-func (rw *RemoteWrite) Run(errChan chan<- error) {
+func (rw *Write) Run(errChan chan<- error) {
```

Comment:
Nit: using `rw` instead of `w` is counter-intuitive and not immediately clear we are dealing with the `Write` struct. Same goes for `rr` instead of `r` for `Read`.

Reply:
that is a good point, will change them.